src/app/modules/item/pages/group-progress-grid/group-progress-grid.component.ts

```diff
@@ -138,12 +141,22 @@ export class GroupProgressGridComponent implements OnChanges, OnDestroy {
   }
 
   showProgressDetail(target: HTMLElement, userProgress: TeamUserProgress, row: DataRow, col: DataColumn): void {
+    if (!this.itemData) {
+      throw new Error('Unexpected: Missed item data');
+    }
     if (!this.group) {
       throw new Error('Unexpected: Missed group');
     }
     this.progressOverlay = {
       target,
       progress: userProgress,
+      ...(this.currentFilter !== 'Groups' && col.type === 'Task' ? {
```

Comment:
I would not make it optional as your "!group && task" filtering is just one part of the filtering you have to do (you filter on perm etc afterwards). So either you put it always or filtering on the only case you need it... I could go for the first option.